internal/adapter/lsp/document.go

```diff
@@ -228,17 +228,22 @@ func (d *document) DocumentLinks() ([]documentLink, error) {
 			})
 		}
 
+		// extract link paths from [title](path) patterns
 		for _, match := range markdownLinkRegex.FindAllStringSubmatchIndex(line, -1) {
-			// Ignore embedded image, e.g. ![title](href.png)
-			if match[0] > 0 && line[match[0]-1] == '!' {
+            // Ignore embedded images ![title](file.png) and tripple dash file 
+            // URIs [title](file:///file.go)
+			if (match[0] > 0 && line[match[0]-1] == '!') ||
+				(match[0] > 0 && line[match[3]+8] == '/') {
```

Comment:
this is a very over-eager match
this matches the **/** here: file:/**/**/, but it would also match the following: ./dir1/file

Also, I don't think this is a good way to use the result of `FindAllStringSubmatchIndex`
- For the `markdownLinkRegex` (`\[([^\]]+?[^\\])\]\((.+?[^\\])\)`), it returns a slice of int slices
    - Each slice within the slice corresponds to a match
    - Each match will contain exactly 6 integers:
        - the starting index of the entire match (inclusive)
        - the ending index of the entire match (exclusive)
        - the starting index of the *bracket* capture group (inclusive)
        - the ending index of the *bracket* capture group (exclusive)
        - the starting index of the *parenthesis* capture group (inclusive)
        - the ending index of the *parenthesis* capture group (exclusive)
    - `match[3]` then corresponds to the index ending the _bracket_ capture group, which seems odd, since we're interested in the *parenthesis* capture group
        - `match[4]` is the one we should use 

Thus, I think we should match against the entire [scheme](https://en.wikipedia.org/wiki/File_URI_scheme)--`file://`--starting at `match[4]`. This is probably best done in a helper function, first checking that `match[5] - match[4]` is large enough for `file://` (8, since we need at least one character after the scheme to be a valid URI), then iterating from `match[4]` to `match[5]` matching each character along the way. If any character doesn't match, we return a `false` early.

Also, I know we're not explicitly supporting windows, but this will cause an obvious portability issue:
- POSIX has the third slash in `file:///` correspond to an actual file: the root file
- Windows seems to append an additional `/`, since `file:///c:` corresponds to the C drive, and `file:///` corresponds to nothing

I don't think we necessary need to support `file://` on Windows with this PR, but I am curious as to the result of this on Windows

Reply:
Ahhh. I thought the ints followed the sequence in which they were found... as in 0 -> n from `[` -> `)`.
I'll have a proper look through all this in the coming days hopefully. But I'm on holiday for a month from Thursday!

Thanks so much for all the detail and thoroughness! 🙏

Reply:
np. it took a lot of testing for me to figure out what exactly `FindAllStringSubmatchIndex` does. It's documentation leaves a lot to be desired

Reply:
@mcDevnagh 

Would this work, or is it in an ok direction?

```go
package main

import (
	"regexp"
)

const testCase = "[example](file:///foo.go)"

var markdownLinkRegex = regexp.MustCompile(`\[([^\]]+?[^\\])\]\((.+?[^\\])\)`)
var fileURIregex = regexp.MustCompile(`file:///`)

func main() {

	var result string
	for _, match := range markdownLinkRegex.FindAllStringSubmatchIndex(testCase, -1) {
		if match[5]-match[4] >= 8 {
			// get the file url portion, existing within the parentheses
			linkURL := testCase[match[4]:match[5]]
			// limit to first 8 characters as per 'file:///' scheme
			result = linkURL[:8]
		}
		// check if the first 8 characters of the URL match 'file:///'
		if fileURIregex.MatchString(result) {
			// if they match, we continue, meaning we DO NOT consider it an
			// internal note url, thus avoiding the lsp error that the note doesn't exist
			continue
		}
	}

}
```

Reply:
I added this approach into document.go and tested. waiting for review.